airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_incremental.py

```diff
@@ -148,6 +151,64 @@ def construct_latest_state_from_messages(messages: List[AirbyteMessage]) -> Dict
     return latest_per_stream_by_name
 
 
+def group_records_by_stream(records: Iterable[AirbyteMessage]) -> Dict[str, List[AirbyteMessage]]:
+    """
+    Group records by stream name
+    """
+    records_by_stream = defaultdict(list)
+    for record in records:
+        records_by_stream[record.record.stream].append(record)
+    return records_by_stream
+
+
+def naive_diff_records(records_1: List[AirbyteMessage], records_2: List[AirbyteMessage]) -> DeepDiff:
+    """
+    Naively diff two lists of records by comparing their data field.
+    """
+    records_1_data = [record.record.data for record in records_1]
+    records_2_data = [record.record.data for record in records_2]
+
+    # ignore_order=True because the order of records in the list is not guaranteed
+    diff = DeepDiff(records_1_data, records_2_data, ignore_order=True)
+    return diff
+
+
+def is_comparable(obj: dict) -> bool:
+    """
+    Checks if the object is comparable (i.e. has __lt__ and __gt__ methods).
+    """
+    return hasattr(obj, "__lt__") and hasattr(obj, "__gt__")
+
+
+def create_cursor_field_parser(stream_config: dict) -> JsonSchemaHelper:
+    """
+    Create a JsonSchemaHelper for the cursor field of a stream.
+    """
+    helper = JsonSchemaHelper(schema=stream_config.stream.json_schema)
+    cursor_field = helper.field(stream_config.cursor_field)
+    return cursor_field
+
+
+def is_cursor_testable(stream_config: dict, example_record: AirbyteMessage) -> bool:
+    """
+    Not all streams have cursor fields that are comparable.
+
+    For example, a cursor field that is a date is comparable, but a cursor field that is a
+    UUID is not.
+
+    Also, some streams do not have cursor fields at all and use a source defined cursor which is
+    meant to be opaque to the platform and should not be relied on.
+    """
+    source_defined_cursor = stream_config.stream.source_defined_cursor
+
+    if not source_defined_cursor:
+        return True
+
+    cursor_field_helper = create_cursor_field_parser(stream_config)
+    record_value = cursor_field_helper.parse(record=example_record.record.data)
+    return is_comparable(record_value)
```

Comment:
I'm not even sure we should try to compare cursor value. As you correctly stated this in your PR description: it's the connector role to define and control the cursor logic, we've no guarantee that what looks lower or greater than from the outside is not overridden inside the connector.
The `compare_cursor_with_threshold` method looks like a best effort things that could eventually break with an exoctic connector implementation. Keeping it would mean that, in case of failure, we'd  likely investigate deeply again to circle back to the same conclusion you come with in this PR.